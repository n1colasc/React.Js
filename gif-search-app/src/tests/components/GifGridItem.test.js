import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe("Pruebas sobre GifGridItem", () => {
  const title = "Titulo de prueba";
  const url = "https:localhost/Prueba.jpg";
  console.log(title);
  const wrapper = shallow(<GifGridItem title={title} url={url} />);
  test("Mostrar componente GifGridItem", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de tener un pÃ¡rrafo con el title", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("Debe de tener la imagen igual al url y alt de los props", () => {
    const img = wrapper.find("img");
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('Debe tener card animate__animated animate__fadeIn', () =>{

    const div = wrapper.find('div');
    const value = div.prop('className');

    expect(value.endsWith('animate__fadeIn')).toBe(true)

  })
});
