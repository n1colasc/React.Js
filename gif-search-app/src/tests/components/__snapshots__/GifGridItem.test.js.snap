// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre GifGridItem Mostrar componente GifGridItem 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo de prueba"
    src="https:localhost/Prueba.jpg"
  />
  <p>
    Titulo de prueba
  </p>
</div>
`;
